sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -491,10 +493,14 @@ private[hive] class HiveMetastoreCatalog(val client: ClientInterface, hive: Hive
     result.copy(expectedOutputAttributes = Some(metastoreRelation.output))
   }
 
-  override def getTables(databaseName: Option[String]): Seq[(String, Boolean)] = {
+  override def getTables(databaseName: Option[String]): Seq[(String, Boolean)] = {hive
     val db = databaseName.getOrElse(client.currentDatabase)
+    client.listTables(db).map(tableName => getTable(db, tableName))
```

Comment:
@dilipbiswal  Hello, are you still working on this problem? I 'm still interested in this problem and  think it is  this part that causes SPARK-6618 test failed. Cause the test is about concurrent acces to metastore, after client.listTables gets all the tables,  then DROP TABLE action may occur,  and then run getTable will thrown NosuchTable Exception.